ci/ray_ci/pipeline/gap_filling_scheduler.py

```diff
@@ -0,0 +1,66 @@
+import subprocess
+from datetime import datetime, timedelta
+from typing import List, Dict, Optional, Any
+
+from pybuildkite.buildkite import Buildkite
+
+
+BRANCH = "master"
+
+
+class GapFillingScheduler:
+    """
+    This buildkite pipeline scheduler is responsible for scheduling gap filling builds
+    when the latest build is failing.
+    """
+
+    def __init__(
+        self,
+        buildkite_organization: str,
+        buildkite_pipeline: str,
+        buildkite_access_token: str,
+        days_ago: int = 1,
+    ):
+        self.buildkite_organization = buildkite_organization
+        self.buildkite_pipeline = buildkite_pipeline
+        self.buildkite = Buildkite()
+        self.buildkite.set_access_token(buildkite_access_token)
+        self.days_ago = days_ago
+
+    def _get_latest_commit_for_build_state(self, build_state: str) -> Optional[str]:
+        latest_commits = self._get_latest_commits()
+        builds = sorted(
+            [
```

Comment:
maybe convert this to a more conventional loop? I always find functional language style comprehension hard to read.. maybe it is just that my math is not that good..